### PR TITLE
Added welcome screen

### LINKED ISSUE #827: Show a welcome background when there are no instances
### Role

I- i don't know anymore i just want to help improve it; well i use the launcher

### Suggestion

I want there to be shown a "Right click here to create your first instance and get started" background text whenever there are no instances to show.

### Benefit

so newbies can know how to get started using the launcher faster

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

_No response_

### PR DESCRIPTION
**Pull request by Trial97**
fixes #827

### CODE REVIEW

**Comment by MbtMarco** [comment]
A little too rough, it should be improved aesthetically in my opinion, but it is still a nice addition.
In my opinion it would be better if the background is not so contrasting and the text is a bit smaller. 
~Also I would put the launcher logo at the top instead of that ( i ).~
_(Personal opinion of a regular user :P)_

**Comment by TheKodeToad** [comment]
Personally I'd just have "No instances :("
Maybe too vague though lol

**Comment by Trial97** [comment]
Yes I know is a crude implementation(is hard to paint with code).
The background stuff is because of the theme.
The size of the text and the text itself can be updated.
The current implementation reflects what I found in the mentioned issue.

**Comment by TheKodeToad** [comment]
> Personally I'd just have "No instances :(" Maybe too vague though lol

Oh, that's very ironic as this is literally based on my mockup :woman_facepalming:

**Comment by TheKodeToad** [comment]
Idk. I think I find the icon weird :face_with_spiral_eyes:

**Comment by Trial97** [comment]
<details>
  <summary>Removed Icon and made text smaller</summary>

image

</details>

File: launcher/ui/instanceview/InstanceView.cpp [hunk 1]
```diff
@@ -536,11 +536,42 @@ void InstanceView::paintEvent(QPaintEvent* event)
 #endif
     option.widget = this;
 
+    if (model()->rowCount() == 0) {
+        painter.save();
+        const QString line1 = "Welcome!";
+        const QString line2 = "Add an instance to get started.";
```
Comment (TheKodeToad) [inline launcher/ui/instanceview/InstanceView.cpp]: Is this even translated?

Comment (Trial97) [reply launcher/ui/instanceview/InstanceView.cpp]: now yes

**Comment by ghost** [comment]
Why is the background solid black? The contrast is definitely too much.

**Comment by TheKodeToad** [comment]
It's the theme

**Comment by ghost** [comment]
This text doesn't actually tell one what they need to do to add an instance, so it's kind of useless

**Comment by Trial97** [comment]
> This text doesn't actually tell one what they need to do to add an instance, so it's kind of useless

Right now I'm open to any suggestions, if you can provide a better language for that or a mockup of how the page should look like I would try to implement it

**Comment by ghost** [comment]
Maybe something like "Right click anywhere to create an instance.". The text should be only a little bit lighter than the background, just enough so to be visible. It should also be vertically centered.

**Comment by TheKodeToad** [comment]
> This text doesn't actually tell one what they need to do to add an instance, so it's kind of useless

There's a fairly obvious button. Also lowering contrast is a bad idea imo.

**Comment by ghost** [comment]
This text is supposed to be a subtle hint, not a screaming reminder. In all software such a text is present, it doesn't contrast so much with the background it's on. Making it so big and so bright would be a definite UX issue.

**Comment by TheKodeToad** [comment]
i don't think so. since it's only visible with no instances, it has no effect on useability.

image

if you take clion as an example, there's no dimming in the default dark mode

I feel like making the text harder to read on purpose is not a good idea lol

**Comment by Trial97** [comment]
Also please note this screen will be visible only if the number of instances is zero. 
And the contrast is from the theme I use.

**Review by TheKodeToad — APPROVED** [review]

**Comment by TheKodeToad** [comment]
image
I just remembered this. maybe it can be improved in the future though

**Review by DioEgizio — COMMENTED** [review]
i think this is more useful.

also i think it's too big and the text should be smaller

File: launcher/ui/instanceview/InstanceView.cpp [hunk 2]
```diff
@@ -478,6 +479,38 @@ void InstanceView::paintEvent([[maybe_unused]] QPaintEvent* event)
 #endif
     option.widget = this;
 
+    if (model()->rowCount() == 0) {
+        painter.save();
+        const QString line1 = tr("Welcome!");
+        const QString line2 = tr("Add an instance to get started.");
```
Comment (DioEgizio) [inline launcher/ui/instanceview/InstanceView.cpp]:
```suggestion
        const QString line2 = tr("Click "Add Instance" to get started.");
```

**Review by DioEgizio — COMMENTED** [review]
i didn't add that to the review. oops

File: launcher/ui/instanceview/InstanceView.cpp (hunk 2 above)
Comment (Trial97) [reply launcher/ui/instanceview/InstanceView.cpp]: should be fixed now .
Also made the text a little smaller.

**Review by DioEgizio — APPROVED** [review]
looks kinda bad but better than not having anything ig

**Review by TayouVR — APPROVED** [review]
considering how annoying it is to work within this code, LGTM

(whenever we revamp the instanceview fully (#30) maybe this could get changed to look nicer, be centered vertically, etc.)

**Comment by TheKodeToad** [comment]
@RokeJulianLockhart can you comment instead of just downvoting?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PrismLauncher/PrismLauncher/pull/1374#issuecomment-1771537503

@TheKodeToad, what do you refer to?

**Comment by TheKodeToad** [comment]
this:
image

**Comment by RokeJulianLockhart** [comment]
> https://github.com/PrismLauncher/PrismLauncher/pull/1374#issuecomment-1772583542

I know that. I ask which comment you refer to, @TheKodeToad.

**Pull request merged**